src/domains/queries.ml

```diff
@@ -195,6 +199,9 @@ struct
     | MustTermAllLoops -> (module MustBool)
     | IsEverMultiThreaded -> (module MayBool)
     | TmpSpecial _ -> (module ML)
+    | MayBeOutOfBounds _ -> (module ID)
+    | MayOverflow _ -> (module MayBool)
+    | AllocMayBeOutOfBounds _ -> (module VDQ.ProdID)
```

Comment:
For the relational VLA oob check, I don't check if the index can be negative because if the relational domain knows that the index is not negative, the bounds for the array index would be improved anyway in the Valuedomain.

However, this does not work for pointer arithmetic. The relational domain doesn't track the relationship between pointers (yet). Therefore the analysis can't give boundaries for the memory access. In the (newly) added test case [85 14](https://github.com/FungOliver/analyzer/blob/b51131f77c4f4ad0c257cc0db1745e67788e6828/tests/regression/85-relational-malloc/14-pointer-tracking-global2.c) memOutOfBounds would raise a warning that we may access the array before the allocated memory. With the new pointer_tracking enabled, which tracks relational relationships between pointers, we know that this cannot be the case. We, therefore, return a tuple of ~`ID`~ `FlatBool` to know if we might access before or after the allocated memory.